File: packages/app/src/services/software/definitions/melonDs.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'melonds';
	public name = 'melonDS';
	public category = [SoftwareCategory.Emulation, SoftwareCategory.Gaming];
	public downloadName = 'melonDS-windows-x86_64.zip';
	public isArchive = true;
	public icon = 'melonds.png';
	public homepage = 'https://melonds.kuribo64.net';

	public async resolveDownloadUrl() {
		return ok('https://melonds.kuribo64.net/downloads/melonDS-windows-x86_64.zip');
	}
}

File: src/lib/software/melonds.py
from src.lib.software import BaseSoftware

class MelonDs(BaseSoftware):
    def __init__(self):
        super().__init__()

        self.download_name = 'melonDS-windows-x86_64.zip'

    def resolve_download_url(self):
        self.url_resolved.emit('https://melonds.kuribo64.net/downloads/melonDS-windows-x86_64.zip')
